### PR TITLE
TDI-38055 - Fix salesforce proxy test to work in container

### PR DESCRIPTION
**Pull request by francisu**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-38055

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/506/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/506/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/509/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/509/)

File: components-common/src/main/java/org/talend/components/common/runtime/ProxyPropertiesRuntimeHelper.java [hunk 1]
```diff
@@ -36,10 +36,6 @@ public ProxyPropertiesRuntimeHelper(ProxyProperties proxySetting) {
             proxyPort = proxySetting.port.getStringValue();
             proxyUser = proxySetting.userPassword.userId.getStringValue();
             proxyPwd = proxySetting.userPassword.password.getStringValue();
-
-            // use socks as default like before
-            SocketAddress addr = new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort));
```
Comment (jzhao-talend) [inline components-common/src/main/java/org/talend/components/common/runtime/ProxyPropertiesRuntimeHelper.java]: If remove this part of code, then the migration job which set socket proxy in salesforce proxy.
Then those configuration would set as http proxy to connection.

**Comment by wwang-talend** [comment]
Some information about this issue : 
https://jira.talendforge.org/browse/TDI-38194?focusedCommentId=461167&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-461167

**Comment by wwang-talend** [comment]
https://jira.talendforge.org/browse/TDI-20972

**Comment by wwang-talend** [comment]
After checking the whole history, get the conclusion : 
For bulk mode, we already use http proxy for the proxy setting in advanced setting.
For basic mode, at the beginning, we use socks for the proxy setting in advanced setting, but after some api upgrading like axis to axis2, salesforce upgrade, or even jvm upgrade, some issue appear like the api is used wrongly above which i show, the the socks proxy can't work for a long time.

**Comment by wwang-talend** [comment]
Agree with zhaojin's opinion, we can keep it for the product. But from my private view, i like http more than socks for the proxy setting.

**Comment by gonchar-ivan** [comment]
Rejected the original issue https://jira.talendforge.org/browse/TDI-38055, so rejecting this PR too
It is not relevant anymore.

**Pull request closed without merging**